src/hooks/useFilter.tsx

```diff
@@ -0,0 +1,64 @@
+import { selectTaskBoard, setFilteredTasks, setTasks } from '@/redux/features/taskBoardSlice'
+import store from '@/redux/store'
+import { TaskResponse } from '@/types/dto/tasks.dto'
+import { AssigneeType, FilterOptions, FilterOptionsKeywords, IFilterOptions } from '@/types/interfaces'
+import { Task } from '@prisma/client'
+import { useEffect, useState } from 'react'
+import { useSelector } from 'react-redux'
+
+const FilterFunctions = {
+  [FilterOptions.ASSIGNEE]: filterByAssignee,
+  [FilterOptions.KEYWORD]: filterByKeyword,
+  [FilterOptions.TYPE]: filterByType,
+}
+
+function filterByAssignee(filteredTasks: TaskResponse[], filterValue: string | null) {
+  const assigneeId = filterValue
+  filteredTasks = !assigneeId
+    ? filteredTasks
+    : assigneeId === 'none'
+      ? filteredTasks.filter((task) => !task.assigneeId)
+      : filteredTasks.filter((task) => task.assigneeId == assigneeId)
+  return filteredTasks as TaskResponse[]
+}
+function filterByKeyword(filteredTasks: TaskResponse[], filterValue: string) {
+  const keyword = (filterValue as string).toLowerCase()
+  filteredTasks =
+    filterValue !== ''
+      ? filteredTasks.filter(
+          (task) => task.title?.toLowerCase().includes(keyword) || task.body?.toLowerCase().includes(keyword),
+        )
+      : filteredTasks
```

Comment:
This ternary's a little tough to read, can we split it out into a function or something?